lib/galaxy/tool_util/verify/interactor.py

```diff
@@ -1290,7 +1290,7 @@ def verify_tool(
             if job_output_exceptions:
                 job_data["output_problems"] = [util.unicodify(_) for _ in job_output_exceptions]
                 status = "failure"
-            if tool_execution_exception:
+            if tool_execution_exception and not expected_failure_occurred:
```

Comment:
I think it might be good to record this anyway ? Can we change the global decision over the test passing / failing later ? (I assume we base that on job_data["execution_problem"] being non-empty ?)

Reply:
> I think it might be good to record this anyway ?

I'm not entirely sure what "this" refers to.

> Can we change the global decision over the test passing / failing later ? (I assume we base that on job_data["execution_problem"] being non-empty ?)

Should we just modify setting of the status based on `expected_failure_occurred`?

I do not have a good understanding of the code. Could you detail "later"? 

To me it seems that planemo bases the decision based on the status, but I'm not sure.

Most importantly I have problems to understand what is causing this. I was considering https://github.com/galaxyproject/galaxy/pull/10304, but as far as I see the output validaton code is not executed if parameter validation errors occur.

Reply:
> I'm not entirely sure what "this" refers to.

`job_data["execution_problem"] = util.unicodify(tool_execution_exception)` which you're skipping now. If this fixes something for you you'll want to check out why, then work on that part